varinfo/variable.py

```diff
@@ -341,17 +341,53 @@ class VariableFromDmr(VariableBase, AttributeContainerFromDmr):
 
     """
 
+    def __init__(
+        self,
+        element: ET.Element,
+        cf_config: CFConfig,
+        namespace: str,
+        full_name_path: str,
+        all_dimensions_sizes: dict[str, int],
+    ):
+        self.all_dimensions_sizes = all_dimensions_sizes
+        super().__init__(element, cf_config, namespace, full_name_path)
+
     def _get_data_type(self, variable: ET.Element) -> str:
         """Extract a string representation of the variable data type."""
         return variable.tag.lstrip(self.namespace).lower()
 
     def _get_shape(self, variable: ET.Element) -> tuple[int]:
-        """Extract the shape of the variable data array. This is not yet
-        implemented as the Dimension information is currently unavailable
-        to the Variable XML content.
-
+        """Extract the shape of the variable data array. First explore
+        XML for - DIM child elements with a size
+        attribute - true for HDF5 files and anonymous dimensions
+        (note no name). If found (with or without names), use them. If
+        not found, use all_dimensions_sizes dictionary with matching dimension
+        names to define the variable shape. Note the all_dimensions_sizes
+        dictionary is filled when parsing xml for variables, capturing
+        dimensions names and sizes for dimension entries - as found in
+        NetCDF files with named dimensions.
         """
 
+        # Retrieve the shape from the dimension size in the
+        # <Dim size=xx/> element.
+        shape = [
+            int(dim_size.attrib['size'])
+            for dim_size in variable.findall(f'.//{self.namespace}Dim[@size]')
+        ]
+
+        # Retrieve the shape from dimension size in the
+        # <Dimension size=xx/> element
+        if (
+            len(shape) != len(self.all_dimensions_sizes)
+            and len(self.all_dimensions_sizes) > 0
```

Comment:
This seems off:

`self.all_dimension_sizes` is a cache of all dimensions for the DMR, not a measure of the total number of dimensions of the variable being evaluated. What we're trying to check here is: does the `shape` list have an element for every dimension of the variable array.

The check should be updated to check whether the length of `shape`, which is the number of `<Dim /> elements with a `size` attribute, is the same as the total number of `<Dim />` elements for the variable.

Reply:
Nice. Marking conversation thread as resolved.